pkg/apis/sources/v1alpha1/ping_types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"knative.dev/pkg/apis"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:defaulter-gen=true
+
+// PingSource is the Schema for the PingSources API.
+type PingSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PingSourceSpec   `json:"spec,omitempty"`
+	Status PingSourceStatus `json:"status,omitempty"`
+}
+
+// TODO: Check that PingSource can be validated and can be defaulted.
+
+var (
+	// Check that it is a runtime object.
+	_ runtime.Object = (*PingSource)(nil)
+
+	// Check that we can create OwnerReferences to a PingSource.
+	_ kmeta.OwnerRefable = (*PingSource)(nil)
+
+	// Check that PingSource can return its spec untyped.
+	_ apis.HasSpec = (*PingSource)(nil)
+)
+
+type PingRequestsSpec struct {
+	ResourceCPU    string `json:"cpu,omitempty"`
+	ResourceMemory string `json:"memory,omitempty"`
+}
+
+type PingLimitsSpec struct {
+	ResourceCPU    string `json:"cpu,omitempty"`
+	ResourceMemory string `json:"memory,omitempty"`
+}
+
+type PingResourceSpec struct {
+	Requests PingRequestsSpec `json:"requests,omitempty"`
+	Limits   PingLimitsSpec   `json:"limits,omitempty"`
+}
+
+// PingSourceSpec defines the desired state of the PingSource.
+type PingSourceSpec struct {
+	// Schedule is the cronjob schedule.
+	// +required
+	Schedule string `json:"schedule"`
+
+	// Data is the data posted to the target function.
+	Data string `json:"data,omitempty"`
+
+	// Sink is a reference to an object that will resolve to a uri to use as the sink.
+	Sink *duckv1.Destination `json:"sink,omitempty"`
+
+	// ServiceAccoutName is the name of the ServiceAccount that will be used to run the Receive
+	// Adapter Deployment.
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// Resource limits and Request specifications of the Receive Adapter Deployment
+	Resources PingResourceSpec `json:"resources,omitempty"`
+}
```

Comment:
I wish we could get rid of both ServiceAccountName and Resources. Users should not have to worry about it.

Reply:
Will do.

Reply:
we can cheat a little because it is a new object, we have 5 weeks until PingSource becomes a support issue